### ISSUE TITLE
Crash when dealing with very large response body

### ISSUE THREAD

**Issue by jiaxian-lin** [EMResearch/EvoMaster#449]
```
Exception in thread "main" java.lang.OutOfMemoryError: Required array length 2147483638 + 2778 is too large
        at java.base/jdk.internal.util.ArraysSupport.hugeLength(ArraysSupport.java:649)
        at java.base/jdk.internal.util.ArraysSupport.newLength(ArraysSupport.java:642)
        at java.base/java.lang.AbstractStringBuilder.newCapacity(AbstractStringBuilder.java:257)
        at java.base/java.lang.AbstractStringBuilder.ensureCapacityInternal(AbstractStringBuilder.java:229)
        at java.base/java.lang.AbstractStringBuilder.append(AbstractStringBuilder.java:740)
        at java.base/java.lang.StringBuilder.append(StringBuilder.java:233)
        at org.glassfish.jersey.message.internal.ReaderWriter.readFromAsString(ReaderWriter.java:153)
        at org.glassfish.jersey.message.internal.ReaderWriter.readFromAsString(ReaderWriter.java:137)
        at org.glassfish.jersey.message.internal.AbstractMessageReaderWriterProvider.readFromAsString(AbstractMessageReaderWriterProvider.java:94)
        at org.glassfish.jersey.message.internal.StringMessageProvider.readFrom(StringMessageProvider.java:54)
        at org.glassfish.jersey.message.internal.StringMessageProvider.readFrom(StringMessageProvider.java:36)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.invokeReadFrom(ReaderInterceptorExecutor.java:233)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$TerminalReaderInterceptor.aroundReadFrom(ReaderInterceptorExecutor.java:212)
        at org.glassfish.jersey.message.internal.ReaderInterceptorExecutor.proceed(ReaderInterceptorExecutor.java:132)
        at org.glassfish.jersey.message.internal.MessageBodyFactory.readFrom(MessageBodyFactory.java:1072)
        at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:885)
        at org.glassfish.jersey.message.internal.InboundMessageContext.readEntity(InboundMessageContext.java:819)
        at org.glassfish.jersey.client.ClientResponse.readEntity(ClientResponse.java:298)
        at org.glassfish.jersey.client.InboundJaxrsResponse$1.call(InboundJaxrsResponse.java:93)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:365)
        at org.glassfish.jersey.client.InboundJaxrsResponse.runInScopeIfPossible(InboundJaxrsResponse.java:244)
        at org.glassfish.jersey.client.InboundJaxrsResponse.readEntity(InboundJaxrsResponse.java:90)
        at org.evomaster.core.problem.rest.service.AbstractRestFitness.handleRestCall(AbstractRestFitness.kt:362)
        at org.evomaster.core.problem.rest.service.BlackBoxRestFitness.doCalculateCoverage(BlackBoxRestFitness.kt:60)
        at org.evomaster.core.problem.rest.service.BlackBoxRestFitness.doCalculateCoverage(BlackBoxRestFitness.kt:16)
        at org.evomaster.core.search.service.FitnessFunction.calculateIndividualWithPostHandling(FitnessFunction.kt:104)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage(FitnessFunction.kt:56)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage$default(FitnessFunction.kt:48)
        at org.evomaster.core.search.algorithms.RandomAlgorithm.searchOnce(RandomAlgorithm.kt:25)
        at org.evomaster.core.search.service.SearchAlgorithm.search(SearchAlgorithm.kt:66)
        at org.evomaster.core.Main$Companion.run(Main.kt:383)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:147)
        at org.evomaster.core.Main$Companion.main(Main.kt:71)
        at org.evomaster.core.Main.main(Main.kt)

**Comment by arcuri82** [comment]
thx for reporting this issue. the problem here is that your API is responding with a payload that has more than 2 __B__illions elements when read as an array of characters. so big that it crashes the JVM, as strings are bound by up to 2^31 elements. 

still, we should not crash EvoMaster, and rather skip such payload with a warning message.

is the payload some kind of very large data file?

**Comment by arcuri82** [comment]
This issue should now be fixed in the new 1.4.0 release
